src/runtime/hot.ts

```diff
@@ -20,3 +21,8 @@ export type Runnable = () => void;
 
 export type Predicate<T> = (input: T) => boolean;
 
+
+
+// Higher Order Data Types
+
+export type Literal = object;
```

Comment:
I'd like to see some jsdoc for these explaining rationale, etc.

Reply:
If Literal is intended to be the base type for things that come out of toLiteral(), you'll need to use an interface rather than a type alias, because type alias information isn't maintained past parse.

Reply:
Good shout. Added docs for the whole `hot.ts` file.